slack_emojinator.py

```diff
@@ -0,0 +1,237 @@
+				uploaded += 1
+			else:
+				print("Skipping {}. Emoji already exists".format(emoji_name))
+				skipped += 1
+		else:
+			upload_emoji(session, emoji_name, filename)
+			print("{} uploaded.".format(filename))
+			uploaded += 1
+	print('\nUploaded {} emoji. ({} already existed)'.format(uploaded, skipped))
+
+
+def export(session, args):
+	if not os.path.exists(args.directory):
+		os.makedirs(args.directory)
+
+	resp = session.get(session.url)
+	tree = lxml.html.fromstring(resp.text)
+	urls = tree.xpath(r'//td[@headers="custom_emoji_image"]/span/@data-original')
+	names = [u.split('/')[-2] for u in urls]
+
+	for emoji_name, emoji_url in zip(names, urls):
+		if "alias" not in emoji_url:  # this does not seem necessary ...
+			file_extension = emoji_url.split(".")[-1]
+			request = session.get(emoji_url, stream=True)
+			if request.status_code == 200:
+				filename = '%s/%s.%s' % (args.directory, emoji_name,
+										 file_extension)
+				with open(filename, 'wb') as out_file:
+					shutil.copyfileobj(request.raw, out_file)
+				del request
+
+
+def delete(session, args):
+	existing_emoji = get_current_emoji_list(session)
+	deleted = 0
+
+	S =[]
+	for path in args.files:
+		patt = re.compile(re.escape(path).replace(r"\*", ".*")+"$")
+		for emoji in existing_emoji:
+			if patt.match(emoji):
+				S.append(emoji)
+
+	if len(S) == 0:
+		print("No emoji matched the given names. Aborting...")
+		return 
+
+	print("Emoji to delete: %s\n"%('; '.join(S)))
+	print("Do you want to delete these %d emoji?"%len(S), end=' ')
+
+	if not confirmation():
+		print("Aborted")
+		return
+
+	for emoji in S:
+		print("Processing {}.".format(emoji))
+
+		if delete_emoji(session, emoji):
+			deleted += 1
+			print("{} deleted.".format(emoji))
+		else:
+			print("Error while deleting {}.".format(emoji))
+	print('\nDeleted {} emoji. ({} failed)'.format(deleted, len(S)-deleted))
+
+
+
+
+
+
+def get_current_emoji_list(session):
+	r = session.get(session.url)
+	r.raise_for_status()
+	x = re.findall("data-emoji-name=\"(.*?)\"", r.text)
+	return x
+
+
+def upload_emoji(session, emoji_name, filename):
+	# Fetch the form first, to generate a crumb.
+	r = session.get(session.url)
+	r.raise_for_status()
+	soup = BeautifulSoup(r.text, "html.parser")
+	crumb = soup.find("input", attrs={"name": "crumb"})["value"]
+
+	data = {
+		'add': 1,
+		'crumb': crumb,
+		'name': emoji_name,
+		'mode': 'data',
+	}
+	files = {'img': open(filename, 'rb')}
+	r = session.post(session.url, data=data, files=files, allow_redirects=False)
+	r.raise_for_status()
+	# Slack returns 200 OK even if upload fails, so check for status of 'alert_error' info box
+	if b'alert_error' in r.content:
+		soup = BeautifulSoup(r.text, "html.parser")
+		crumb = soup.find("p", attrs={"class": "alert_error"})
+		print("Error with uploading %s: %s" % (emoji_name, crumb.text))
+
+def delete_emoji(session, emoji_name):
+	data = {
+		'name': emoji_name,
+		'token': "xoxs-195008312805-195741066343-195722307958-36ca7f8642",
```

Comment:
Er, you probably didn't want to commit this.

Reply:
cc @DaniJVaz

Reply:
crap, that was not supposed to be public at all. Thanks for catching it, even though I somehow didn't get the notification at all.

Reply:
@DaniJVaz make sure you change your token, too, eh?